kinder/pkg/test/workflow/workflow.go

```diff
@@ -137,11 +148,77 @@ func NewWorkflow(file string) (*Workflow, error) {
 		}
 	}
 
-	// TODO: detect and resolve includes by expanding the top level Workflow
-
 	return &w, nil
 }
 
+// expandImports imports a secondary workflow into the top level Workflow
+func (w *Workflow) expandImports(file string) error {
+	tasks := w.Tasks
+	w.Tasks = Tasks{}
+	for i, t := range tasks {
+		// check if the task does not defines an import, preserve it as it is
+		if t.Import == "" {
+			w.Tasks = append(w.Tasks, t)
+			continue
+		}
+
+		// otherwise it is an import task
+		// ensure the import task does not have other settings
+		if t.Dir != "" {
+			return errors.Errorf("invalid workflow file %s: task #%d - dir setting can't be combined with import directive", file, i+1)
+		}
+		if t.Cmd != "" {
+			return errors.Errorf("invalid workflow file %s: task #%d - cmd setting can't be combined with import directive", file, i+1)
+		}
+		if len(t.Args) != 0 {
+			return errors.Errorf("invalid workflow file %s: task #%d - args setting can't be combined with import directive", file, i+1)
+		}
+		if t.Force != false {
+			return errors.Errorf("invalid workflow file %s: task #%d - force setting can't be combined with import directive", file, i+1)
+		}
+		if t.Timeout != 0 {
+			return errors.Errorf("invalid workflow file %s: task #%d - timeout setting can't be combined with import directive", file, i+1)
+		}
+
+		// reads the Import file
+		// TODO: implement a check for avoinding circular imports
+		path, _ := filepath.Abs(t.Import)
+		wx, err := NewWorkflow(path)
+		if err != nil {
+			return errors.Wrapf(err, "error importing workflow file %s", path)
+		}
+
+		// merge the vars from the import file into the parent file
+		// in case of conflicts, vars in the parent file will shadow vars in the import file
+		for k, v := range wx.Vars {
+			if _, ok := w.Vars[k]; !ok {
+				w.Vars[k] = v
+				continue
+			}
+			log.Warnf("var %s in workflow file %s is shadowed by var %[1]s in parent workflow file %[3]s", k, path, file)
+		}
+
+		// merge the env vars from the import file into the parent file
+		// in case of conflicts, env vars in the parent file will shadow env vars in the import file
+		for k, v := range wx.Env {
+			if _, ok := w.Env[k]; !ok {
+				w.Env[k] = v
+				continue
+			}
+			log.Warnf("env var %s in workflow file %s is shadowed by env var %[1]s in parent workflow file %[3]s", k, path, file)
+		}
+
+		// import all tasks from the import file into the parent file, removing task name prefix
+		re := regexp.MustCompile(`^task\-\d{2}\-?`)
+		for _, tx := range wx.Tasks {
+			tx.Name = re.ReplaceAllString(tx.Name, "")
+			w.Tasks = append(w.Tasks, tx)
```

Comment:
is it adding the tasks of the imported workflow at the bottom of the already existing tasks of the parent workflow?

Reply:
It is replacing the import task with tasks in the imported workflow (not adding at the bottom)
 e.g.

workflow A
- Task 1 
- Task 2 import workflow B
- Task 3 

And

workflow B
- Task 1 
- Task 2 

Will became
- Task 1 (from A/task1)
- Task 2 (from B/task1)
- Task 3 (from B/task2)
- Task 4 (from A/task2)

Reply:
You can think at this as an include

Reply:
thanks for the example
/lgtm